flumine/backtest/simulated.py

```diff
@@ -185,10 +197,25 @@ def _process_price_matched(
                 else:
                     _size_matched = avail["size"]
                 _matched = [publish_time, avail["price"], round(_size_matched, 2)]
-                self._update_matched(_matched)
+                _matches.append(_matched)
             else:
                 break
 
+        if self.order.order_type.time_in_force == "FILL_OR_KILL":
+            if size_remaining == 0:
+                self._update_multi_matched(lst_data=_matches)
+            elif self.order.order_type.min_fill_size:
+                filled_size = sum(x[2] for x in _matches)
+                if filled_size >= self.order.order_type.min_fill_size:
+                    self._update_multi_matched(lst_data=_matches)
+        else:
+            self._update_multi_matched(lst_data=_matches)
+
+    def _update_multi_matched(self, lst_data: List) -> None:
```

Comment:
Test?

Reply:
There are no direct tests for `_update_multi_matched`, but it is indirectly tested by:

- test__process_price_matched_back_fill_or_kill_succeeds
- test__process_price_matched_back_fill_or_kill_no_matches
- test__process_price_matched_back_fill_or_kill_min_fill_size_succeeds
- test__process_price_matched_back_fill_or_kill_min_fill_size_no_matches